sphinx_rtd_theme/__init__.py

```diff
@@ -36,6 +37,11 @@ def extend_html_context(app, pagename, templatename, context, doctree):
      # Add ``sphinx_version_info`` tuple for use in Jinja templates
      context['sphinx_version_info'] = sphinx_version
 
+     # Populate the context with all the READTHEDOCS variables
+     for variable in os.environ:
+         if variable.startswith('READTHEDOCS'):
+             context[variable] = os.environ.get(variable)
+
```

Comment:
I wonder if we should undeprecate [the `canonical_url` option](https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-canonical_url) and use `READTHEDOCS_CANONICAL_URL` ?

(not for this PR, just in case you agree, we can open a separate issue)

Reply:
I don't think so. I'd say the theme should probably define `base_url = os.environ("READTHEDOCS_CANONICAL_URL")` -- which will make everything work properly

Reply:
Yes, except for DirHTML, but we're on it :)